specs/multi-url-support.md

```diff
@@ -0,0 +1,112 @@
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | -------- | ----- | --- |
+| 1.0 | 2022-12-09 | Initial specifications | @sebastienlevert
+
+
+## Configuration file
+
+A configuration file should be available for developers who want to support scenarios outside of Microsoft Graph. The file should be at the root of the proxy working folder and named `config.json`. This file should have default RegEx values for all supported Microsoft Graph clouds and SharePoint URLs.
+
+This configuration can also be extended by "any" URLs to support 1P endpoints, vanity URLs, custom services, etc.
+
+| Property | Type | Comments |
+| ------- | -------- | ----- |
+| id | String (unique) | This can be any string value but needs to be unique |
+| regEx | String | String value of the RegEx to use |
+| isWildcard | Boolean | If the value is `true`, the RegEx gets an appended wildcard token. If `false`, only the RegEx value should be considered |
+
+> **Note**
+> These RegEx need to be validated in the context of the App
+
+```json
+{
+  "endpoints": [
+    { 
+      "id": "microsoftGraph",
+      "regEx": "https:\/\/(canary.)?((graph|dod-graph).microsoft.(com|us)|microsoftgraph.chinacloudapi.cn)",
+      "isWildcard": true
+    },
+    {
+      "id": "sharePoint",
+      "regEx": "https:\/\/(.*).sharepoint(-df)?.(com|us)",
+      "isWildcard": true
+    },
+    {
+      "id": "customService",
+      "regEx": "https:\/\/customService.azurewebsites.net",
+      "isWildcard": true
+    },
+    {
+      "id": "customService2",
+      "regEx": "https:\/\/customService2.azurewebsites.net/api/GetMyData",
+      "isWildcard": false
+    }
+  ]
+    
+  }
+}
+```
+
+## Remove the `--cloud` option on the Proxy
+
+This new capability renders the `--cloud` option obsolete on the Proxy. We should deprecate the capability, remove associated documentation and once the Proxy hits its next major version, introduce the breaking change. Until then, if the option is used, notify the developer that this option now is obsolete and will use the `config.json` file as its source of truth for URLs it should proxy.
+
+## Mocking responses
+
+Responses can be mocked and should be updated to include either the absolute URL for the response to be returned or a RegEx. This RegEx would be used in scenarios where a developer want to test the same set of responses on multiple clouds or on different environments. Is is possible to use the `id` property of the endpoint to reuse the base RegEx with using `{valueOfTheId}` and append paths.

```

Comment:
Following the simplified implementation of URLs to intercept, I suggest we use the existing wildcard mechanism, and if users want to provide a single mock for multiple Graph API URLs, they use for example `https://*graph*/...` which should capture all Graph URLs across the different clouds. Depending on the need, we can provide recommended examples in the documentation.